### PR TITLE
[osx] - add support for Xcode 8

### PR DESCRIPTION
**Pull request by Memphiz**
This allows compiling with Xcode8 on either OSX 10.11 (El Capitan) or macOS 10.12 (Sierra).

thx @davilla for giving a hand.

@ksooo if you are still on el capitan - might want to test this out?

### CODE REVIEW

**Comment by ksooo** [comment]
> @ksooo if you are still on el capitan - might want to test this out?

 sorry, updated to sierra yesterday

**Comment by koying** [comment]
Sure, mention-bot, I'm totally the king of xcode :)

**Comment by Memphiz** [comment]
@koying maybe you did something in depends configura.ac at some point ;)

**Comment by FernetMenta** [comment]
also on sierra already

File: tools/depends/native/config.site.native.in [hunk 1]
```diff
@@ -24,3 +24,12 @@ fi
 
 LD_LIBRARY_PATH=@prefix@/@tool_dir@/lib:$LD_LIBRARY_PATH
 NASM=@prefix@/@tool_dir@/bin/yasm
+
+if test "@platform_os@" = "osx" ; then
+  # Xcode 8 + 10.11, clock_gettime and getentropy is present
+  # in 10.12 but will get wrongly detected if building on 10.11
+  ac_cv_search_clock_gettime=no
```
Comment (FernetMenta) [inline tools/depends/native/config.site.native.in]: maximum respect! that was beyond my skills as I tried.

Comment (Memphiz) [inline tools/depends/native/config.site.native.in]: Yep passthrough this respect to @davilla :)

**Comment by xhaggi** [comment]
I'm on el capitan but can't test it before tomorrow.

**Comment by Memphiz** [comment]
Well since i need this now that i have updated to xcode8 - lets ask jenkins build this please anyway and if @xhaggi has some sparetime left to test this in the next days - glad to have another ok (beside that building with xcode8 on 10.11 is only intermediate as most people will switch to sierra, when 12.1 comes out latest :) ).

File: tools/depends/native/config.site.native.in (hunk 1 above)
Comment (MrMC) [inline tools/depends/native/config.site.native.in]: The test needs to get expanded to include ios as the same issue is present there. At least for tvOS. Have not tried under iOS yet.

Comment (Memphiz) [inline tools/depends/native/config.site.native.in]: i am about to try it for ios - but for ios we also need to bump deployment target (5.1 won't work anymore) so i will defer the related PR to after krypton branching ...

**Comment by MrMC** [comment]
just a note, a build/run on sierra without this should work, but run that same build on el capitan and below and it will fail to run because of missing symbols.

File: tools/depends/native/config.site.native.in (hunk 1 above)
Comment (MrMC) [inline tools/depends/native/config.site.native.in]: anything below 8 is deadbeef in my book :)

**Comment by Memphiz** [comment]
yep that makes sense too :)

File: tools/depends/native/config.site.native.in (hunk 1 above)
Comment (Memphiz) [inline tools/depends/native/config.site.native.in]: btw this very PR seems to compile fine with xcode8 and ios 10 sdk (deployment target bumped to 6.0) ... not runtime tested yet though

**Comment by fetzerch** [comment]
Fails for me with xcode7 on 10.11:
```
server-tools/instance-manager/Makefile.am:103: `%'-style pattern rules are a GNU make extension
/usr/bin/clang -I. -fheinous-gnu-extensions -no-cpp-precomp -arch x86_64 -mmacosx-version-min=10.8  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -isystem /Users/Shared/xbm
c-depends/macosx10.11_x86_64-target/include  -DHAVE_CONFIG_H -fheinous-gnu-extensions -no-cpp-precomp -arch x86_64 -mmacosx-version-min=10.8  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/
MacOSX10.11.sdk -isystem /Users/Shared/xbmc-depends/macosx10.11_x86_64-target/include  -ggdb3 -Wall -W   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -fPIC -MT ne
ttle-internal.o -MD -MP -MF nettle-internal.o.d -c nettle-internal.c \
        && true
../lib/replace/replace.c:811:23: error: unknown type name 'clockid_t'; did you mean 'clock_t'?
int rep_clock_gettime(clockid_t clk_id, struct timespec *tp)
                      ^~~~~~~~~
                      clock_t
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/sys/_types/_clock_t.h:30:33: note: 'clock_t' declared here
typedef __darwin_clock_t        clock_t;
                                ^
1 error generated.
```

**Comment by Memphiz** [comment]
@fetzerch thx will look into it tomorrow :)

**Comment by Jalle19** [comment]
Travis CI seems to support XCode 8 too, at least according to https://docs.travis-ci.com/user/osx-ci-environment/. Might be worth updating `.travis.yml` while you're at it.

**Comment by Memphiz** [comment]
@Jalle19 i don't see that travis builds for osx at all. So its not done just by "updating" something. No time to dive into travis-ci to add osx support to it.

**Comment by Jalle19** [comment]
Oh, I was mistaken since we do that for binary addons.

**Comment by Memphiz** [comment]
jenkins had the same issue @fetzerch and i was able to reproduce aswell. PR updated - new testbuild succeeded:

http://jenkins.kodi.tv/job/OSX-64/9309/

**Comment by fetzerch** [comment]
looks good here as well

**Pull request merged**